### PR TITLE
fix: Same default environment

### LINKED ISSUE #2411: sessions sent without an environment when environment option is not set
### Description

If the environment is not set for the SDK errors will default to production. However, sessions are sent without an environment. 

Expected outcome - sessions would be sent with an environment
Actual outcome: sessions are sent without an environment set

Steps To Reproduce

1. set the environment and release options
2. Send an error event and session
3. Observe that session and error event with the correct environment
4. repeat without setting the environment option
5. Observe the error event with an environment of production
6. Observe the session with no environment

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

SentryOptions is the source of truth for the default environment, and is used on events and sessions.

## :bulb: Motivation and Context

Closes #2411

## :green_heart: How did you test it?

With a debug proxy to see the actual values getting sent, and with unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [ ] No breaking changes -> Do we classify this as a breaking change, or just a bug fix? 🤔

### CODE REVIEW

**Comment by kevinrenskers** [comment]
I first ran the sample app on the main branch, only removing the `options.environment = "test-app"` line, using a debug proxy (tip: [Proxyman](https://proxyman.io) is great). I triggered an error and I can see that both the event and the session have the environment `debug`, as that's what's being set in the sample app (`scope.setEnvironment("debug")`). 

I then also removed that line, and ran the sample app again, triggering an error. The event get environment "production" while the session gets no environment.

Now I did the same using my branch. First only removing `options.environment = "test-app"`. Now the event gets the environment "production" but the session gets the environment "debug". This should both be "debug" I assume, as scope.setEnvironment("debug")` is still set. If I remove that line, then both event and session get "production".

So obviously I broke something, where the scope's environment is now overridden by the options's environment. I should be able to figure that out. But I am also wondering why the session's environment property is publicly writable? This means that you can send different environment values for events and sessions, on purpose. Why?

File: Sources/Sentry/SentryClient.m [hunk 1]
```diff
@@ -546,12 +545,6 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         event.dist = dist;
     }
 
-    NSString *environment = self.options.environment;
```
Comment (kevinrenskers) [inline Sources/Sentry/SentryClient.m:549]: I don't know why this bit code was here, to first set the event's environment, then do some other work, then set the event's environment again some lines later? It seems completely unnecessary at least, with the current changes where `self.options.environment` now always has a value.

**Comment by brustolin** [comment]
@philipphofmann do you know why `SentrySession` is public? I did not found a single public API that uses the session class.

**Comment by philipphofmann** [comment]
> @philipphofmann do you know why `SentrySession` is public? I did not found a single public API that uses the session class.

I guess no good reason. I made it internal https://github.com/getsentry/sentry-cocoa/pull/2451.

**Pull request merged**